File: david/sessions/2018/easy/66_plus_one.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        carry = 0
        plus = 1
        l = len(digits)
        for i in reversed(xrange(l)):
            digits[i] += plus + carry
            plus = 0
            carry = digits[i] / 10
            digits[i] %= 10
        if carry > 0:
            digits.insert(0, carry)
        return digits

File: pia/2020/easy/66_plus_one.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function (digits) {
  digits[digits.length - 1]++;

  let carry = 0;
  for (let i = digits.length - 1; i >= 0; i--) {
    const sum = digits[i] + carry;

    if (sum === 10) {
      digits[i] = sum - 10;
      carry = 1;
    } else {
      digits[i] = sum;
      carry = 0;
    }
  }
  if (carry > 0) {
    digits.unshift(carry);
  }
  return digits;
};
